pkg/allocation/converters/converter.go

```diff
@@ -206,28 +210,14 @@ func convertAllocationPortsV1Alpha1ToAgonesPortsV1(in []*pb.AllocationResponse_G
 }
 
 // convertStateV1ToAllocationStateV1Alpha1 converts GameServerAllocationState V1 (GSA) to AllocationResponse_GameServerAllocationState
-func convertStateV1ToAllocationStateV1Alpha1(in allocationv1.GameServerAllocationState) pb.AllocationResponse_GameServerAllocationState {
+func convertStateV1ToError(in allocationv1.GameServerAllocationState) error {
 	switch in {
 	case allocationv1.GameServerAllocationAllocated:
-		return pb.AllocationResponse_Allocated
+		return nil
 	case allocationv1.GameServerAllocationUnAllocated:
-		return pb.AllocationResponse_UnAllocated
+		return status.Error(codes.ResourceExhausted, "there isn't any available Game Server to allocate")
 	case allocationv1.GameServerAllocationContention:
-		return pb.AllocationResponse_Contention
+		return status.Error(codes.Aborted, "too many concurrent request has overwhelmed the system")
 	}
-	return pb.AllocationResponse_Unknown
-}
-
-// convertAllocationStateV1Alpha1ToStateV1 converts AllocationResponse_GameServerAllocationState to GameServerAllocationState V1 (GSA)
-// if the conversion cannot happen, it returns false.
-func convertAllocationStateV1Alpha1ToStateV1(in pb.AllocationResponse_GameServerAllocationState) (allocationv1.GameServerAllocationState, bool) {
-	switch in {
-	case pb.AllocationResponse_Allocated:
-		return allocationv1.GameServerAllocationAllocated, true
-	case pb.AllocationResponse_UnAllocated:
-		return allocationv1.GameServerAllocationUnAllocated, true
-	case pb.AllocationResponse_Contention:
-		return allocationv1.GameServerAllocationContention, true
-	}
-	return allocationv1.GameServerAllocationUnAllocated, false
+	return status.Error(codes.Unknown, "Unknown issue")
```

Comment:
I think here should be also lower case to be consistent with above
```suggestion
	return status.Error(codes.Unknown, "unknown issue")
```

Reply:
Done